### PR TITLE
fix: wallet connect multisig signing

### PR DESCRIPTION
**Pull request by Asmadek**

### CODE REVIEW

**Comment by Asmadek** [comment]
Task linked: [CU-8693122fp Multisig Operations. Wrong address is sent to NovaWallet.](https://app.clickup.com/t/8693122fp)

**Comment by Asmadek** [comment]
Task linked: [CU-8693122uw Multisig Operations. Is not call reconnect, if WalletConnect connection has been lost](https://app.clickup.com/t/8693122uw)

**Comment by Asmadek** [comment]
Task linked: [CU-869311rg2 Multisig Operations signing. Wrong text on sign button](https://app.clickup.com/t/869311rg2)

**Review by egor0798 — APPROVED** [review]

File: src/renderer/entities/transaction/lib/common/errors.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import type { ObjectValues } from '@renderer/shared/core';
+
+export const OperationError = {
+  INVALID_FEE: 'staking.notEnoughBalanceForFeeError',
+  INVALID_DEPOSIT: 'staking.notEnoughBalanceForDepositError',
+  EMPTY_ERROR: ' ', // error for invalid state with no hint text
+} as const;
+
+export type OperationErrorType = ObjectValues<typeof OperationError>;
```
Comment (tuul-wq) [inline src/renderer/entities/transaction/lib/common/errors.ts:9]: Nice!

File: src/renderer/entities/transaction/lib/index.ts [hunk 2]
```diff
@@ -4,3 +4,6 @@ export * from './common/utils';
 export * from './common/types';
 export * from './common/constants';
 export * from './validateBalance';
+
+export { OperationError } from './common/errors';
```
Comment (tuul-wq) [inline src/renderer/entities/transaction/lib/index.ts:8]: Found out that you can do this instead - `export { OperationError, type OperationErrorType } from './common/errors';`

File: src/renderer/features/operation/sign/lib/constants.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export enum ReconnectStep {
```
Comment (tuul-wq) [inline src/renderer/features/operation/sign/lib/constants.ts:1]: Pure enums pollute bundle with worthless code
```suggestion
export const enum ReconnectStep {
```

File: src/renderer/features/operation/sign/lib/utils.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import { ReconnectStep } from './constants';
+
+export const isReconnectingStep = (step: ReconnectStep) => step === ReconnectStep.RECONNECTING;
```
Comment (tuul-wq) [inline src/renderer/features/operation/sign/lib/utils.ts:3]:
```suggestion
export const isReconnectingStep = (step: ReconnectStep): boolean => step === ReconnectStep.RECONNECTING;
```

File: src/renderer/features/operation/sign/model/wallet-connect-sign-model.ts [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+  to: walletConnectModel.events.connect,
+});
+
+sample({
+  clock: walletConnectModel.events.connected,
+  source: {
+    step: $reconnectStep,
+    session: walletConnectModel.$session,
+  },
+  filter: ({ step, session }) => isReconnectingStep(step) || (isConnectedStep(step) && Boolean(session?.topic)),
```
Comment (tuul-wq) [inline src/renderer/features/operation/sign/model/wallet-connect-sign-model.ts:44]: I like this `filter`
Could we somehow add `Boolean(session?.topic)` inside utility function?

File: src/renderer/features/operation/sign/model/wallet-connect-sign-model.ts [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+  fn: ({ session }) => {
```
Comment (tuul-wq) [inline src/renderer/features/operation/sign/model/wallet-connect-sign-model.ts:45]:
```suggestion
  fn: ({ session }) => session?.topic!
```

File: src/renderer/features/operation/sign/model/wallet-connect-sign-model.ts [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    return session?.topic!;
+  },
+  target: walletConnectModel.events.currentSessionTopicUpdated,
+});
+
+sample({
+  clock: combineEvents({
+    events: [reconnectStarted, walletConnectModel.events.sessionTopicUpdateDone, walletConnectModel.events.connected],
```
Comment (tuul-wq) [inline src/renderer/features/operation/sign/model/wallet-connect-sign-model.ts:53]: I wonder, it actually requires completion of all listed events?

File: src/renderer/features/operation/sign/model/wallet-connect-sign-model.ts [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+      newAccounts: updatedAccounts,
+    };
+  },
+  target: walletConnectModel.events.accountsUpdated,
+});
+
+sample({
+  clock: walletConnectModel.events.connectionRejected,
+  source: $reconnectStep,
+  filter: (step) => step === ReconnectStep.RECONNECTING,
```
Comment (tuul-wq) [inline src/renderer/features/operation/sign/model/wallet-connect-sign-model.ts:88]: We have this one `isReconnectingStep(...)` 😉
short way `filter: isReconnectingStep`, but check is it working

File: src/renderer/features/operation/sign/ui/WalletConnect/WalletConnect.tsx [hunk 9]
```diff
@@ -30,66 +33,40 @@ export const WalletConnect = ({
 
   const session = useUnit(walletConnectModel.$session);
   const client = useUnit(walletConnectModel.$client);
+  const reconnectStep = useUnit(walletConnectSignModel.$reconnectStep);
 
   const chains = chainsService.getChainsData();
 
   const [txPayload, setTxPayload] = useState<Uint8Array>();
   const [unsignedTx, setUnsignedTx] = useState<UnsignedTransaction>();
-  const [isNeedUpdate, setIsNeedUpdate] = useState<boolean>(false);
```
Comment (tuul-wq) [inline src/renderer/features/operation/sign/ui/WalletConnect/WalletConnect.tsx:38]: 👍

File: src/renderer/features/operation/sign/ui/WalletConnect/WalletConnect.tsx [hunk 10]
```diff
@@ -139,14 +110,18 @@ export const WalletConnect = ({
             transactionPayload: unsignedTx,
           },
         },
-      });
+      };
+
+      const result = await client.request<{
```
Comment (tuul-wq) [inline src/renderer/features/operation/sign/ui/WalletConnect/WalletConnect.tsx:115]: Won't work without specifying generic?

File: src/renderer/features/operation/sign/ui/WalletConnect/WalletConnect.tsx [hunk 11]
```diff
@@ -167,27 +142,35 @@ export const WalletConnect = ({
   const walletName = session?.peer.metadata.name || t('operation.walletConnect.defaultWalletName');
 
   const getStatusProps = () => {
-    if (isReconnectingModalOpen) {
+    if (isReconnectingStep(reconnectStep)) {
```
Comment (tuul-wq) [inline src/renderer/features/operation/sign/ui/WalletConnect/WalletConnect.tsx:145]: 👍 Looks clean

**Review by tuul-wq — CHANGES_REQUESTED** [review]
Overall it's a good PR 👍

File: src/renderer/features/operation/sign/model/wallet-connect-sign-model.ts (hunk 7 above)
Comment (Asmadek) [reply src/renderer/features/operation/sign/model/wallet-connect-sign-model.ts:51]: Right

**Review by tuul-wq — APPROVED** [review]

**Pull request merged**